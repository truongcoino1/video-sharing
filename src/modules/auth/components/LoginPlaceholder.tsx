import React, { memo } from "react";
import Skeleton from "react-loading-skeleton";

const LoginPlaceholder = (): JSX.Element => {
  return (
    <div className="flex login-placeholder">
      <Skeleton containerClassName="mr-12" className="w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px]  h-[40px] flex-shrink-0 overflow-hidden relative" />
      <Skeleton containerClassName="mr-12" className="w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px] mr-12 h-[40px] flex-shrink-0 overflow-hidden relative" />
      <Skeleton className="w-[80px] min-w-[80px] lg:w-[134px] lg:min-w-[150px] h-[40px] flex-shrink-0 overflow-hidden relative" />
    </div>
  );
};

export default memo(LoginPlaceholder);
