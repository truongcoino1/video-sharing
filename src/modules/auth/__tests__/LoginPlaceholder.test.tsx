import { render, cleanup } from "@testing-library/react";
import LoginPlaceholder from "../components/LoginPlaceholder";

describe("LoginPlaceholder", () => {
  afterEach(cleanup);

  it("share should render without crash", () => {
    const result = render(<LoginPlaceholder />);
    const elm = result.container.querySelector(".login-placeholder");
    expect(elm).toBeInTheDocument();
  });

  it("should render correct LoginPlaceholder", () => {
    const element = render(<LoginPlaceholder />);
    expect(element).toMatchSnapshot();
  });
});
