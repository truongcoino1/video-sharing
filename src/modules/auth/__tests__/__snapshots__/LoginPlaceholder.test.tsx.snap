// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPlaceholder should render correct LoginPlaceholder 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex login-placeholder"
      >
        <span
          aria-busy="true"
          aria-live="polite"
          class="mr-12"
        >
          <span
            class="react-loading-skeleton w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px]  h-[40px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
          class="mr-12"
        >
          <span
            class="react-loading-skeleton w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px] mr-12 h-[40px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-[80px] min-w-[80px] lg:w-[134px] lg:min-w-[150px] h-[40px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex login-placeholder"
    >
      <span
        aria-busy="true"
        aria-live="polite"
        class="mr-12"
      >
        <span
          class="react-loading-skeleton w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px]  h-[40px] flex-shrink-0 overflow-hidden relative"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
        class="mr-12"
      >
        <span
          class="react-loading-skeleton w-[100px] min-w-[100px] lg:w-[200px] lg:min-w-[200px] mr-12 h-[40px] flex-shrink-0 overflow-hidden relative"
        >
          ‌
        </span>
        <br />
      </span>
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-[80px] min-w-[80px] lg:w-[134px] lg:min-w-[150px] h-[40px] flex-shrink-0 overflow-hidden relative"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
